### ISSUE TITLE
Replicate all the experiments for Parity and First

### ISSUE THREAD

**Issue by giacomocamposampiero** [giacomocamposampiero/palindrome-transformer#11]
Replicate exact encoder experiments:

### FIRST exact
- [x] Create datasets for FIRST with string lengths in the range [0 : 100 : 10] and [0 : 1000 : 100].
- [ ] Add a line in training that divides the total cross entropy by the number of training examples to get the cross entropy per bit.
- [x] Run experiment with the following settings: no normalization, normalization with a couple of epsilon values (they tried 0 and 1e-5 in the paper)

### PARITY exact
- [x] Create datasets for PARITY with string lengths in the range [0 : 10 : 1] and [0 : 100 : 10]
- [ ] Add a line in training that divides the total cross entropy by the number of training examples to get the cross entropy per bit.
- [x] Run experiment with the following settings: no normalization, normalization with a couple of epsilon values (they tried 0 and 1e-5 in the paper)
- [ ] Optional: Additional experiments to plot the cross entropy and accuracy depending on value of W_V parameter (Figure 3 in the paper)

Replicate learned encoder experiments:
- [x] FIRST transformer trained from scratch on shorter strings (n = 10,30,100,300) and tested on longer strings (n = 1000); trained on 100 epochs with 100 training string (test also more, as in the original paper they're randomly generated and therefore the encoder get to see a lot more unique strings); test w/ and w/o scaled attention
- [x] PARITY transformer trained from scratch using different parameters (no indications in the paper)
